StoryboardInstantiable.swift

```diff
@@ -8,12 +8,25 @@
 
 import UIKit
 
+/** 
+ Applicable to UIViewController classes which have a visual representation
+ within a storyboard. Define the name of the corresponding storyboard and
+ clients will be able to instantiate the UIViewController without having to
+ know the details of it's storyboard.
+ */
 protocol StoryboardInstantiable: class {
+    
+    /// The name of the storyboard this viewcontroller's layout is defined
     static var storyboardName: String { get }
 }
 
 extension StoryboardInstantiable where Self: UIViewController {
-
+    
+    /**
+     Initializes the UIViewController from the context of its sortyboard
+     
+     - Returns: A new UIViewController instance
```

Comment:
Also important to state that this UIViewControllers' storyboardIdentifier (in the storyboard file) should be the same as its class name